modules/lti/src/main/java/org/opencastproject/lti/LtiServlet.java

```diff
@@ -218,13 +222,28 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
     // The URL of the LTI tool. If no specific tool is passed we use the test tool
     UriBuilder builder;
     try {
-      String customTool = URLDecoder
-              .decode(StringUtils.trimToEmpty(req.getParameter(LTI_CUSTOM_TOOL)), StandardCharsets.UTF_8.displayName());
-      customTool = customTool.replaceAll(
-          "/?ltitools/(?<tool>[^/]*)/index.html\\??",
-          "/ltitools/index.html?subtool=${tool}&"
-      );
-      URI toolUri = new URI(customTool);
+      String messageType = StringUtils.trimToEmpty(req.getParameter(LTI_MESSAGE_TYPE));
+      URI toolUri;
+
+      if (messageType.equals(CONTENT_ITEM_MESSAGE_TYPE)) {
+        toolUri = new URI(URLDecoder.decode(StringUtils.trimToEmpty(
+                req.getParameter(LTI_CUSTOM_DL_TOOL)), "UTF-8"));
+      } else if (req.getRequestURI().startsWith("/lti/player/")) {
+        String mpID = req.getRequestURI().replace("/lti/player/", "");
+        String redirectUrl = "/engage/theodul/ui/core.html?id=" + mpID + "&ltimode=true";
```

Comment:
redirect here just to "/play/"+mpID or preferably, use the same redirect code that the /play/ID service uses to redirect to the configured player. We don't want to hardcode a URL for a specific player here.